sycl/unittests/thread_safety/ThreadUtils.h

```diff
@@ -0,0 +1,42 @@
+#include <thread>
+#include <vector>
+
+class ParallelTask;
+
+class Thread {
+public:
+  Thread(ParallelTask *Ptr) : MTask(Ptr) {}
+  void start(size_t id);
+  void wait();
+  void body(size_t id);
+
+private:
+  std::thread MThread;
+  ParallelTask *MTask;
+};
+
+class ThreadPool {
+public:
+  ThreadPool(ParallelTask *p);
+  void initialize(int size);
+  void start();
+  void wait();
+
+private:
+  std::vector<Thread *> MThreadPool;
+  ParallelTask *MTask;
+};
+
+class ParallelTask {
+  friend class ThreadPool;
+
+public:
+  ParallelTask() : MPool(this) {}
+
+  void execute(int threadCount);
+
+  virtual void taskBody(std::size_t id) = 0;
+
+private:
+  ThreadPool MPool;
+};
```

Comment:
What if I'd like to test if things go naughty way when two different functions are run simultaneously? Is there any facility for this kind of scenario?